docs/testing_mbed_OS_5.md

```diff
@@ -0,0 +1,187 @@
+# Testing in mbed OS 5
+
+The way tests are ran and compiled in mbed OS 5 is substantially different than in previous versions of mbed. Previously tests were located in one known location and a python file was kept (`tools/tests.py`) keeping track of their dependencies, capabilities, and configurations. mbed OS 5 has adopted a more distributed approach to testing. Now test code lives alongside the test code, and these are dynamically discovered by the test tools.
```

Comment:
"are run"
"different from previous"
"Previously, tests were"
"Python file (`tools/tests.py`) kept track"
"Test code lives alongside the application code"